templates-pipelines/windows-bios-installer/README.md

```diff
@@ -0,0 +1,100 @@
+2. `create-vm` task creates a VirtualMachine called `windows-bios-installer-*`
+   from the empty DataVolume and with the `windows-bios-installer-cd-rom` DataVolume attached as a CD-ROM.
+   A second DataVolume with the virtio-win ISO will also be attached (Pipeline parameter `virtioContainerDiskName`).
+3. `wait-for-vmi-status` task waits until the VirtualMachine shuts down.
+4. `cleanup-vm` deletes the installer VirtualMachine and ISO DataVolume (also in case of failure of the previous tasks).
+5. The output artifact will be the `win10` DataVolume with the basic Windows installation.
+   It will boot into the Windows OOBE and needs to be setup further before it can be used.
+
+## How to run
+
+Pipeline run with resolver:
+{% for item in pipeline_runs_yaml %}
+```yaml
+oc create -f - <<EOF
+{{ item | to_nice_yaml }}EOF
+```
+{% endfor %}
+
+### Usage in multiple namespaces
+
+When a user defines a different namespace in e.g. `baseDvNamespace`, then the serviceAccount under which the pipeline is running will require additional permissions in that namespace. Required permissions to run task in different namespace can be found in README of each task.
+
+## Possible Optimizations
+
+### Windows Source ISO Caching
+
+Windows source ISO can be downloaded and cached/stored in a DataVolume. So the subsequent PipelineRuns don't have to download the same ISO multiple times.
+
+You can for example create the following DataVolume first and remove create-source-dv step from the windows10-installer pipeline:
+
+```yaml
+ apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: windows10-source
+spec:
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 7Gi
+    volumeMode: Filesystem
+  source:
+    http:
+      url: WIN_IMAGE_DOWNLOAD_URL
+```
+
+## Cancelling/Deleteting pipelineRuns
+
+When running the example pipelines, they create temporary objects (DataVolumes, VirtualMachines, etc.). Each pipeline has its own clean up system which should keep the cluster clean from leftovers. In case user hard deletes or cancels running pipelineRun, the pipelineRun will not clean temporary objects and objects will stay in the cluster and then they have to be deleted manually. To prevent this behaviour, cancel the [pipelineRun gracefully](https://tekton.dev/docs/pipelines/pipelineruns/#gracefully-cancelling-a-pipelinerun). It triggers special tasks, which remove temporary objects and keep only result DataVolume/PVC.
+
+#### Obtaining a download URL in an automated way
+
+The script [`getisourl.py`](https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/pipelines/windows-bios-installer/getisourl.py) can be used to automatically obtain a Windows 10 ISO download URL.
+
+The prerequisites are:
+
+- python3-selenium
+- chromedriver
+- chromium
+
+Run it as follows to initialize a WIN_URL variable.
+
+```bash
+# Real URL can look differently
+WIN_URL=$(./getisourl.py | sed 's/&/\\&/g')
```

Comment:
Use same variable name as above? Also, is the call to `sed` really required? If so, the script should be adapted.

Or it might work better with quotes around the subshell?

Reply:
If you think it requires additional changes, please create an issue, I will work on it after this PR is merged.

Reply:
Will it work like this?

`WIN_URL="$(./getisourl.py)"`

Reply:
It looks like it is working.